bindings/py/tests/algorithms/cells4_test.py

```diff
@@ -205,3 +251,35 @@ def testLearn(self):
       cells.compute(x, True, False)
 
     self._testPersistence(cells)
+
+  def testEquals(self):
+    nCols = 10
+    c1 = createCells4(nCols)
+    c2 = createCells4(nCols)
+    self.assertEquals(c1, c2)
+    
+    # learn
+    data = [numpy.random.choice(nCols, nCols/3, False) for _ in xrange(10)]   
+    for idx in data:
+      x = numpy.zeros(nCols, dtype="float32")
+      x[idx] = 1.0
+      c1.compute(x, True, True)
+      c2.compute(x, True, True)
+      self.assertEquals(c1, c2)
+
+    self.assertEquals(c1, c2)
+
+    c1.rebuildOutSynapses()
+    c2.rebuildOutSynapses()
+    self.assertEquals(c1, c2)
+
+    # inference
+    data = [numpy.random.choice(nCols, nCols/3, False) for _ in xrange(100)]
+    for idx in data:
+      x = numpy.zeros(nCols, dtype="float32")
+      x[idx] = 1.0
+      c1.compute(x, True, False)   
+      c2.compute(x, True, False)
+      self.assertEquals(c1, c2)
+
+    self.assertEquals(c1, c2)
```

Comment:
I suggest you add a negative test. Change something in one cells4 and assert they are not equal.